// @vkontakte/vkui/dist/es6/components/FormStatus/FormStatus.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import classNames from '../../lib/classNames';
import getClassName from '../../helpers/getClassName';
var baseClassName = getClassName('FormStatus');

var FormStatus = function FormStatus(_ref) {
  var state = _ref.state,
      title = _ref.title,
      children = _ref.children,
      className = _ref.className,
      dangerouslySetInnerHTML = _ref.dangerouslySetInnerHTML,
      restProps = _objectWithoutProperties(_ref, ["state", "title", "children", "className", "dangerouslySetInnerHTML"]);

  return React.createElement("div", _extends({}, restProps, {
    className: classNames(baseClassName, _defineProperty({}, "FormStatus--".concat(state), state), className)
  }), title && React.createElement("div", {
    className: "FormStatus__title"
  }, title), dangerouslySetInnerHTML && React.createElement("div", {
    className: "FormStatus__content",
    dangerouslySetInnerHTML: dangerouslySetInnerHTML
  }), children && !dangerouslySetInnerHTML && React.createElement("div", {
    className: "FormStatus__content"
  }, children));
};

FormStatus.propTypes = {
  state: PropTypes.oneOf(['default', 'error']),
  title: PropTypes.node,
  children: PropTypes.node,
  dangerouslySetInnerHTML: PropTypes.object,
  className: PropTypes.string
};
export default FormStatus;
