// @vkontakte/vkui/dist/es6/components/HeaderButton/HeaderButton.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import Tappable from '../Tappable/Tappable';
import getClassName from '../../helpers/getClassName';
import classNames from '../../lib/classNames';
var baseClassName = getClassName('HeaderButton');

var HeaderButton = function HeaderButton(_ref) {
  var className = _ref.className,
      children = _ref.children,
      primary = _ref.primary,
      restProps = _objectWithoutProperties(_ref, ["className", "children", "primary"]);

  var isPrimitive = typeof children === 'string' || typeof children === 'number';
  var component = restProps.href ? 'a' : 'button';
  return React.createElement(Tappable, _extends({}, restProps, {
    component: component,
    activeEffectDelay: 200,
    className: classNames(baseClassName, className, {
      'HeaderButton--primary': primary
    })
  }), isPrimitive ? React.createElement("span", {
    className: "HeaderButton__primitive"
  }, children) : children);
};

HeaderButton.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  primary: PropTypes.bool,

  /**
   * Делает из кнопки ссылку
   */
  href: PropTypes.string
};
HeaderButton.defaultProps = {
  primary: false
};
export default HeaderButton;
