// @vkontakte/vkui/dist/components/FormStatus/FormStatus.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classNames2 = _interopRequireDefault(require("../../lib/classNames"));

var _getClassName = _interopRequireDefault(require("../../helpers/getClassName"));

var baseClassName = (0, _getClassName.default)('FormStatus');

var FormStatus = function FormStatus(_ref) {
  var state = _ref.state,
      title = _ref.title,
      children = _ref.children,
      className = _ref.className,
      dangerouslySetInnerHTML = _ref.dangerouslySetInnerHTML,
      restProps = (0, _objectWithoutProperties2.default)(_ref, ["state", "title", "children", "className", "dangerouslySetInnerHTML"]);
  return _react.default.createElement("div", (0, _extends2.default)({}, restProps, {
    className: (0, _classNames2.default)(baseClassName, (0, _defineProperty2.default)({}, "FormStatus--".concat(state), state), className)
  }), title && _react.default.createElement("div", {
    className: "FormStatus__title"
  }, title), dangerouslySetInnerHTML && _react.default.createElement("div", {
    className: "FormStatus__content",
    dangerouslySetInnerHTML: dangerouslySetInnerHTML
  }), children && !dangerouslySetInnerHTML && _react.default.createElement("div", {
    className: "FormStatus__content"
  }, children));
};

FormStatus.propTypes = {
  state: _propTypes.default.oneOf(['default', 'error']),
  title: _propTypes.default.node,
  children: _propTypes.default.node,
  dangerouslySetInnerHTML: _propTypes.default.object,
  className: _propTypes.default.string
};
var _default = FormStatus;
exports.default = _default;
