// @vkontakte/vkui/dist/es6/components/FormLayoutGroup/FormLayoutGroup.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import getClassName from '../../helpers/getClassName';
import classNames from '../../lib/classNames';
var baseClassNames = getClassName('FormLayoutGroup');

var FormLayoutGroup = function FormLayoutGroup(_ref) {
  var children = _ref.children,
      top = _ref.top,
      bottom = _ref.bottom,
      className = _ref.className,
      restProps = _objectWithoutProperties(_ref, ["children", "top", "bottom", "className"]);

  return React.createElement("div", _extends({
    className: classNames(baseClassNames, className)
  }, restProps), children);
};

FormLayoutGroup.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object,
  top: PropTypes.node,
  bottom: PropTypes.node,
  status: PropTypes.oneOf(['default', 'error', 'valid'])
};
export default FormLayoutGroup;
